applications/ContactStructuralMechanicsApplication/custom_conditions/penalty_frictionless_mortar_contact_condition.cpp

```diff
@@ -66,6 +66,143 @@ PenaltyMethodFrictionlessMortarContactCondition<TDim,TNumNodes, TNormalVariation
 /***********************************************************************************/
 /***********************************************************************************/
 
+template< SizeType TDim, SizeType TNumNodes, bool TNormalVariation, SizeType TNumNodesMaster >
```

Comment:
also this is not recognized as cpp ...?

Reply:
Maybe because is large and it becomes lazy

Reply:
In PenaltyMethodFrictionlessMortarContactCondition, You should check the units involved in the condition in line 111. They are not consistent.

Reply:
Yes you are right, in fact geometry are is in both sides, thanks for notice it